concent_api/core/payments/bankster.py

```diff
@@ -45,3 +74,79 @@ def claim_deposit(
     )
 
     return requestor_has_enough_deposit, provider_has_enough_deposit
+
+
+def finalize_payment(
+    subtask_id: str,
+    concent_use_case: ConcentUseCase,
+    requestor_ethereum_address: str,
+    provider_ethereum_address: str,
+    subtask_cost: int,
+) -> Tuple[ClaimPaymentInfo, ClaimPaymentInfo]:
+    """
+    This operation tells Bankster to pay out funds from deposit.
+    For each claim, Bankster uses SCI to submit an Ethereum transaction to the Ethereum client which then propagates it
+    to the rest of the network.
+    Hopefully the transaction is included in one of the upcoming blocks on the blockchain.
+    """
+
+    assert isinstance(subtask_id, str)
+    assert isinstance(concent_use_case, ConcentUseCase)
+    assert isinstance(requestor_ethereum_address, str)
+    assert isinstance(provider_ethereum_address, str)
+    assert isinstance(subtask_cost, int) and subtask_cost > 0
+
+    assert concent_use_case in [ConcentUseCase.FORCED_ACCEPTANCE, ConcentUseCase.ADDITIONAL_VERIFICATION]
+    assert len(requestor_ethereum_address) == ETHEREUM_ADDRESS_LENGTH
+    assert len(provider_ethereum_address) == ETHEREUM_ADDRESS_LENGTH
+    assert provider_ethereum_address != requestor_ethereum_address
+
+    # Bankster determines the amount that needs to be claimed from each account.
+    requestor_can_pay = min(service.get_deposit_value(client_eth_address=requestor_ethereum_address), subtask_cost)  # pylint: disable=no-value-for-parameter
+    provider_can_pay = (
+        min(
```

Comment:
it doesn't seem right - from what I understood, if provider doesn't have enough funds, UC is aborted, so we should never have a need check this.

Reply:
Right, but in meantime provider's account might be updated, so we should check again he still has something there. I think that preventing this is related with https://github.com/golemfactory/concent/issues/479 .

Reply:
Ok